<!-- Launch file for hmi_support_frobit node -->
<launch>

    <!-- Sensor drivers -->
    <group ns="fmSensors">
        <!-- SICK laser scanner -->
        <node name="sick_tim310s01" pkg="sick_tim" type="sick_tim310s01" respawn="false" output="screen">
            <param name="min_ang" type="double" value="-1.8326" />
            <param name="max_ang" type="double" value="1.8326" />
            <param name="intensity" type="bool" value="True" />
            <param name="skip" type="int" value="0" />
            <param name="frame_id" type="str" value="laser_link" />
            <param name="time_offset" type="double" value="-0.001" />
            <param name="publish_datagram" type="bool" value="False" />
        </node>
    </group>


    <group ns="fmLib">

        <!-- Frobit serial device and protocol interface -->
        <node pkg="serial_nmea" name="frobit_nmea" type="serial_nmea_node" respawn="true" output="screen">
            <param name="nmea_from_device_pub" value="/fmSignal/nmea_from_frobit"/>
            <param name="nmea_to_device_sub" value="/fmSignal/nmea_to_frobit"/>
            <param name="serial_device" value="/dev/frobit"/>
            <param name="serial_baudrate" value="57600"/>
            <param name="use_nmea_checksum" value="true"/>
        </node>

        <!-- Serial comm with Arduino -->
        <!-- <node pkg="rosserial_python" name="rosserial_python" type="serial_node.py" output="screen" args="/dev/ttyACM0" /> --> <!-- TODO udev symlink -->
    </group>


    <group ns="fmControllers">
        <!-- Frobit interface -->
        <node pkg="sdu_frobit_v2" name="frobit" type="frobit_interface_node.py" output="screen">
            <param name="actuation_enable_sub" value="/fmSafe/actuation_enable" />
            <param name="cmd_vel_sub" value="/fmCommand/safe_vel" />
            <param name="enc_left_pub" value="/fmInformation/encoder_left" />
            <param name="enc_right_pub" value="/fmInformation/encoder_right" />
            <param name="wheel_status_left_pub" value="/fmInformation/wheel_status_left" />
            <param name="wheel_status_right_pub" value="/fmInformation/wheel_status_right" />
            <param name="wheel_feedback_left_pub" value="/fmInformation/wheel_feedback_left" />
            <param name="wheel_feedback_right_pub" value="/fmInformation/wheel_feedback_right" />
            <param name="wheel_pid_left_pub" value="/fmInformation/wheel_pid_left" />
            <param name="wheel_pid_right_pub" value="/fmInformation/wheel_pid_right" />
            <param name="nmea_from_frobit_sub" value="/fmSignal/nmea_from_frobit" />
            <param name="nmea_to_frobit_pub" value="/fmSignal/nmea_to_frobit" />
            <param name="publish_wheel_status_rate" value="10" />
            <param name="publish_wheel_feedback_rate" value="10" />
            <param name="publish_wheel_pid_rate" value="10" />
            <param name="castor_front" value="true" />
            <param name="max_linear_acceleration" value="16.2" />
            <param name="max_angular_acceleration" value="16.0" />
            <param name="wheel_drive_feed_forward" value="85" />
            <param name="wheel_drive_kp" value="5" />
            <param name="wheel_drive_ki" value="10" />
            <param name="wheel_drive_kd" value="5" />
            <param name="wheel_drive_max_integral_output" value="300" />
            <param name="wheel_turn_feed_forward" value="110" />
            <param name="wheel_turn_kp" value="10" />
            <param name="wheel_turn_ki" value="20" />
            <param name="wheel_turn_kd" value="3" />
            <param name="wheel_turn_max_integral_output" value="300" />
            <param name="supply_voltage_scale_factor" value="0.03747" />
            <param name="min_supply_voltage" value="12.1" />
            <param name="show_voltage_interval" value="300" />
        </node>
    </group>

    <node pkg="docking_with_walls" name="docking_with_walls_node" type="docking_with_walls_node.py" output="screen" />

</launch>